File: 1537-maximum-score-after-splitting-a-string/1537-maximum-score-after-splitting-a-string.cpp
class Solution {
public:
    int maxScore(string str) {


         int maximumSum = 0;
 
    // To store the total ones
    int totalOnes;
 
    // Count the total numbers of ones
    // in string str
    totalOnes = count(str.begin(),
                      str.end(), '1');
 
    // To store the count of zeros and
    // ones while traversing string
    int zero = 0, ones = 0,n=str.length()-1;


 
    // Iterate the given string and
    // update the maximum sum
    for (int i = 0; i<n; i++) {
 
        if (str[i] == '0') {
            zero++;
        }
        else {
            ones++;
        }
 
        // Update the maximum Sum
        maximumSum
            = max(
                maximumSum,
                zero + (totalOnes - ones));
    }
 
    return maximumSum;
        
    }
};

File: 1537-maximum-score-after-splitting-a-string/1537-maximum-score-after-splitting-a-string.py
class Solution(object):
    def maxScore(self, str):
        """
        :type s: str
        :rtype: int
        """
        maximumSum = 0
 
    # To store the total ones
 
    # Count the total numbers of ones
    # in str
        totalOnes = 0
        for i in str:
            if i == '1':
                totalOnes += 1
 
    # To store the count of zeros and
    # ones while traversing string
        zero = 0
        ones = 0
 
    # Iterate the given and
    # update the maximum sum
        i = 0
        while i < (len(str)-1):
 
            if (str[i] == '0'):
                zero += 1
            else:
                ones += 1
 
        # Update the maximum Sum
            maximumSum= max(maximumSum,zero + (totalOnes - ones))
            i += 1
 
        return maximumSum
        